### PR TITLE
kie-issue#163: In DMN Editor's Boxed Expression Editor, long names are not being truncated in DecisionTable sub-headers and in the name of root expressions

### LINKED ISSUE #163: In DMN Editor's Boxed Expression Editor, long names are not being truncated in DecisionTable sub-headers and in the name of root expressions
**Context**
In the new Boxed Expression Editor (BEE), long expression names are not correctly truncated, causing the size of the cells getting too big and broken the resize hint:

Case 1 of 2: Sub-header in Decision Table
image

Case 2 of 2: top-level expression name:
image

This behavior is already working properly with Literal Expressions and some other cases (see the validated cases at the end):
image
 
**Acceptance Criteria**
1. Long text should be truncated and a "..." must be appended to hint the user that there is hidden text in root expression name;
2. Long text should be truncated and a "..." must be appended to hint the user that there is hidden text in sub-headers;
3. Resize hint should not be broken.

```[tasklist]
### Acceptance Criteria cases
- [x] Literal expression root text truncated + resize hint
- [ ] Context expression root text truncated + resize hint
- [ ] Decision Table root text truncated + resize hint
- [ ] Decision Table sub-headers text truncated + resize hint
- [ ] Relation expression root text truncated + resize hint
- [x] Relation sub-headers text truncated + resize hint
- [ ] Relation expression root text truncated + resize hint
- [ ] Function expression root text truncated + resize hint
- [ ] List expression expression root text truncated + resize hint
```

### PR DESCRIPTION
**Pull request by danielzhe**
Closes: https://github.com/kiegroup/kie-issues/issues/163

Literal:
image

Context:
image

Decision Table:
image

Relation:
image

Invocation:
image

List:
image

Funcion (BKM):
image

### CODE REVIEW

File: packages/boxed-expression-component/src/table/BeeTable/BeeTableThResizable.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ export function BeeTableThResizable<R extends object>({
     cssClasses.push(column.groupType ?? "");
     // cssClasses.push(column.cssClasses ?? ""); // FIXME: Breaking Decision tables because of positioning of rowSpan=2 column headers (See https://github.com/kiegroup/kie-issues/issues/162)
     return cssClasses.join(" ");
-  }, [column, columnKey]);
+  }, [columnKey, column.dataType, column.groupType]);
```
Comment (danielzhe) [inline packages/boxed-expression-component/src/table/BeeTable/BeeTableThResizable.tsx:86]: Fixin deps array.

File: packages/boxed-expression-component/src/table/BeeTable/BeeTableThResizable.tsx [hunk 2]
```diff
@@ -106,53 +106,6 @@ export function BeeTableThResizable<R extends object>({
     return onGetWidthToFitData() + extraSpace;
   }, [onGetWidthToFitData]);
 
-  // const { updateColumnResizingWidths } = useBeeTableResizableColumnsDispatch();
```
Comment (danielzhe) [inline packages/boxed-expression-component/src/table/BeeTable/BeeTableThResizable.tsx:109]: Cleaning

File: packages/boxed-expression-component/src/table/BeeTable/BeeTableThResizable.tsx [hunk 3]
```diff
@@ -168,15 +121,15 @@ export function BeeTableThResizable<R extends object>({
   useEffect(() => {
     function onEnter(e: MouseEvent) {
       e.stopPropagation();
-      setHoverInfo((prev) => getHoverInfo(e, th!));
+      setHoverInfo(() => getHoverInfo(e, th!));
     }
 
     function onMove(e: MouseEvent) {
-      setHoverInfo((prev) => getHoverInfo(e, th!));
```
Comment (danielzhe) [inline packages/boxed-expression-component/src/table/BeeTable/BeeTableThResizable.tsx:175]: Cleaning

**Review by ljmotta — APPROVED** [review]

**Review by yesamer — APPROVED** [review]
Manually tested, it works well. @danielzhe Thanks!

**Pull request merged**